cw_decred/lib/wallet_addresses.dart

```diff
@@ -26,20 +68,42 @@ class DecredWalletAddresses extends WalletAddresses {
     return nAddr;
   }
 
+  List<AddressInfo> getAddressInfos() {
+    if (addressInfos.containsKey(0)) {
+      return addressInfos[0]!;
+    }
+    return <AddressInfo>[];
+  }
+
   List<String> addresses() {
-    final res = libdcrwallet.addresses(walletInfo.name);
-    final addrs = (json.decode(res) as List<dynamic>).cast<String>();
+    final addrs = List<String>.from(usedAddresses);
+    for (final k in addressesMap.keys) {
+      addrs.add(k);
+    }
     return addrs;
   }
 
-  @override
-  set address(String addr) {}
-
-  @override
-  Future<void> init() async {}
-
-  @override
-  Future<void> updateAddressesInBox() async {
+  Future<void> updateAddress(String address, String label) async {
+    if (!addressInfos.containsKey(0)) {
+      return;
+    }
+    addressInfos[0]!.forEach((info) {
+      if (info.address == address) {
+        info.label = label;
+      }
+    });
     await saveAddressesInBox();
   }
+
+  (List<String>, List<String>, int) libAddresses() {
+    final nUsed = "10";
+    final nUnused = "1";
+    final res = libdcrwallet.addresses(walletInfo.name, nUsed, nUnused);
+    final decoded = json.decode(res);
+    final usedAddrs = List<String>.from(decoded["used"]);
+    final unusedAddrs = List<String>.from(decoded["unused"]);
+    // index is the index of the first unused address.
+    final index = decoded["index"] ?? 0;
+    return (usedAddrs, unusedAddrs, index);
```

Comment:
Please consider using a class like this instead, so the code where this method is called will be slightly easier to understand (e.g. using res.usedAddrs instead of res.$1)...

```
class LibAddresses {
  final List<String> usedAddrs, unusedAddrs;
  final int firstUnusedAddrIndex;

  LibAddresses(
      {required this.usedAddrs, required this.unusedAddrs, required this.firstUnusedAddrIndex});
}
```

Reply:
Ok doing this.